src/ra.hrl

```diff
@@ -44,12 +44,18 @@
                           suspended |
                           disconnected.
 
+-type ra_voter_status() :: yes | {no, ra_nonvoter_reason()}.
+
+-type ra_nonvoter_reason() :: #{target := ra_index()}.
+
 -type ra_peer_state() :: #{next_index := non_neg_integer(),
                            match_index := non_neg_integer(),
                            query_index := non_neg_integer(),
                            % the commit index last sent
                            % used for evaluating pipeline status
                            commit_index_sent := non_neg_integer(),
+                           %% whether the peer is part of the consensus
+                           voter := ra_voter_status(),
```

Comment:
`:=` makes this field required in the map, is it required always?

Reply:
I have no strong opinion. My only reasoning was that it is easier to deconstruct args when the field is required.